geofileops/fileops.py

```diff
@@ -760,6 +760,18 @@ def rename_column(
             raise ValueError(f"rename_column not supported for {path}")
 
         # Rename column
+        if column_name.lower() == new_column_name.lower():
+            temp_column_name = Path(tempfile.mktemp()).name
+            while temp_column_name in info.columns:
+                temp_column_name = Path(tempfile.mktemp()).name
```

Comment:
```suggestion
            for index in range(9999):
                temp_column_name = f"tmp_{index}"
                if temp_column_name not in info.columns:
                    break
```
Creating a temporary file for each try feels a bit weird... I think this is equivalent?